core/src/main/java/org/mule/runtime/core/privileged/processor/CompositeProcessorChainRouter.java

```diff
@@ -59,10 +61,16 @@ protected ReactiveProcessor getExecutableFunction() {
 
   private BiFunction<CoreEvent, MessageProcessorChain, CoreEvent> processChain() {
     return (event, processorChain) -> {
+      Latch completionLatch = new Latch();
       BaseEventContext childContext = child((BaseEventContext) event.getContext(), ofNullable(getLocation()));
+      childContext.onTerminated((response, throwable) -> completionLatch.countDown());
       CoreEvent result = from(processWithChildContext(event, processorChain, childContext)).block();
       // Block until all child contexts are complete
-      from(childContext.getCompletionPublisher()).block();
+      try {
+        completionLatch.await();
```

Comment:
no timeout?

Reply:
no

Reply:
there should be a timeout, the shutdown timeout from the configuration in the muleContext